test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py

```diff
@@ -2165,20 +2085,36 @@ def validate(self):
                                 code='invalid-module-deprecation-source',
                                 msg=('The deprecation version for a module must be added in this collection')
                             )
-                            # Treat the module as not to be removed:
-                            raise ValueError('')
-                        removed_in = self.StrictVersion(str(version))
-                    except ValueError:
-                        end_of_deprecation_should_be_removed_only = False
-                    else:
-                        if not self.collection:
-                            strict_ansible_version = self.StrictVersion('.'.join(ansible_version.split('.')[:2]))
-                            end_of_deprecation_should_be_removed_only = strict_ansible_version >= removed_in
-                        elif self.collection_version:
-                            strict_ansible_version = self.collection_version
-                            end_of_deprecation_should_be_removed_only = strict_ansible_version >= removed_in
                         else:
-                            end_of_deprecation_should_be_removed_only = False
+                            removed_in = self.StrictVersion(str(version))
+
+                    except ValueError:
+                        self.reporter.trace(
+                            path=self.object_path,
+                            tracebk=traceback.format_exc()
+                        )
+
+                    if not self.collection:
+                        strict_ansible_version = self.StrictVersion('.'.join(ansible_version.split('.')[:2]))
+                        end_of_deprecation_should_be_removed_only = strict_ansible_version >= removed_in
```

Comment:
This will crash, since `removed_in` has not been defined when the above `except` goes through.